crates/iceberg-catalog/src/service/authz/implementations/openfga/mod.rs

```diff
@@ -0,0 +1,405 @@
+use std::{collections::HashSet, str::FromStr};
+
+use crate::{
+    request_metadata::RequestMetadata,
+    service::{
+        authz::{
+            Authorizer, ErrorModel, ListProjectsResponse, NamespaceAction, ProjectAction, Result,
+            ServerAction, TableAction, ViewAction, WarehouseAction,
+        },
+        token_verification::Actor,
+        NamespaceIdentUuid, TableIdentUuid,
+    },
+    ProjectIdent, WarehouseIdent, CONFIG,
+};
+use openfga_rs::open_fga_service_client::OpenFgaServiceClient;
+use openfga_rs::{
+    tonic::{
+        self,
+        codegen::{Body, Bytes, StdError},
+    },
+    CheckRequest, CheckRequestTupleKey, ConsistencyPreference, ListObjectsRequest,
+};
+
+mod client;
+mod models;
+mod service_ext;
+
+pub use client::{new_bearer_auth_client, new_client_credentials, new_unauthenticated_client};
+pub use models::{CollaborationModelVersion, CollaborationModels};
+pub use service_ext::ClientHelper;
+
+lazy_static::lazy_static! {
+    static ref AUTH_CONFIG: crate::config::OpenFGAConfig = {
+
+        CONFIG.openfga.clone().expect("OpenFGAConfig not found")
+    };
+}
+
+#[derive(Clone, Debug)]
+pub struct OpenFGAAuthorizer<T>
+where
+    T: Clone + Sync + Send + 'static,
+    T: tonic::client::GrpcService<tonic::body::BoxBody>,
+    T::Error: Into<StdError>,
+    T::ResponseBody: Body<Data = Bytes> + Send + 'static,
+    <T::ResponseBody as Body>::Error: Into<StdError> + Send,
+{
+    client: OpenFgaServiceClient<T>,
+    store_id: String,
+    authorization_model_id: String,
+}
+
+#[async_trait::async_trait]
+impl<T> Authorizer for OpenFGAAuthorizer<T>
```

Comment:
Are we expecting other Ts than tonic::transport::Channel here?

Reply:
Yes, interceptors for auth :/